src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java

```diff
@@ -72,21 +67,28 @@ public void setRest(RestOperations rest) {
      *            the id of the event.
      * @return ResponseEntity
      */
-    public ResponseEntity<String> getEventDataById(String eventId) {
-        if(StringUtils.isNotBlank(url)) {
-            final String erUrl = URI.create(url.trim() + "/" + "{id}").normalize().toString();
-            LOGGER.debug("The URL to ER is: {}", erUrl);
-
-            final Map<String, String> params = Collections.singletonMap("id", eventId);
-            LOGGER.trace("The ID parameter is set");
-            try {
-                final ResponseEntity<String> response = rest.getForEntity(erUrl, String.class, params);
+    public ResponseEntity getEventDataById(String eventId) {
+        String erUrl = null;
+
+        try {
+            if(StringUtils.isNotBlank(erBaseUrl)) {
+                request
+                    .setHttpMethod(HttpMethod.GET)
+                    .setBaseUrl(erBaseUrl)
+                    .setEndpoint("{id}")
```

Comment:
What {id}? Shouldn't ER "events" entry point specified. Is it some hidden functionality you use that it not seen in the code?
Since method is called getEventDataById(), I guess same endpoint will be used for every call to this function, so entry point can be hardcoded into code or use a property where entrypoint is specified.

Reply:
It is not something i did come up with. If you look in the old code it looked like this.

`final String erUrl = URI.create(url.trim() + "/" + "{id}").normalize().toString();`

I just added it like it was but translated to the new lib.

Reply:
Ok,, code looks very confusing. Let it be as it is then.
But the endpoint should be "/events/{eventId}"
Looking at this code now, first half of endpoint ,"/events", seems to be included in erBaseUrl.
Which is wrong and confusing, if you ask me.
This can be fixed in another PR then.

Reply:
Ye i agree with that. In order to change that, we will need to make changes in the configuration so that the endpoint is not part of er.url.